src/hawkbit-client.c

```diff
@@ -272,27 +272,29 @@ static size_t curl_write_cb(void *content, size_t size, size_t nmemb, void *data
  * @param[in]  jsonRequestBody    REST request body. If NULL, no body is sent.
  * @param[out] jsonResponseParser REST response
  * @param[out] error              Error
- * @return HTTP Status code (Standard codes: 200 = OK, 524 = Operation timed out, 401 = Authorization needed, 403 = Authentication failed )
+ * @return TRUE if request and response parser (if given) suceeded, FALSE otherwise (error set).
  */
-static gint rest_request(enum HTTPMethod method, const gchar* url, JsonBuilder* jsonRequestBody, JsonParser** jsonResponseParser, GError** error)
+static gboolean rest_request(enum HTTPMethod method, const gchar* url, JsonBuilder* jsonRequestBody, JsonParser** jsonResponseParser, GError** error)
 {
         gchar *postdata = NULL;
         g_autofree gchar *token = NULL;
         RestPayload *fetch_buffer = g_new0(RestPayload, 1);
         glong http_code = 0;
         struct curl_slist *headers = NULL;
         CURL *curl = curl_easy_init();
+        gboolean ret = TRUE;
         if (!curl) {
                 g_set_error(error, RHU_HAWKBIT_CLIENT_CURL_ERROR, -1, "curl init failed");
-                return -1;
+                ret = FALSE;
```

Comment:
The actual changes in this PR are mainly fixing the cleanup handling. But this is not what the commit messages states.
Thus either split this up or describe this explicitly.

Reply:
Right, changing the return type and refactoring of `rest_request()` got mixed up here. Will fix it.